### PR TITLE
Fix Volume material mapping

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
Solve two issue with the volume material mapping :

- When the sampling step size of the volume was larger than the distance between two geantino interaction this would result in an incorrect thickness for the material Step.

- Replace float by double for the accumulated volume material as if the number of step per bin became too large this would result in rounding error.

### CODE REVIEW

File: Core/src/Material/VolumeMaterialMapper.cpp [hunk 1]
```diff
@@ -312,9 +313,7 @@ void Acts::VolumeMaterialMapper::mapMaterialTrack(
         float remainder = rmIter->materialProperties.thickness() -
                           m_cfg.mappingStep * volumeStep;
         properties.scaleThickness(m_cfg.mappingStep / properties.thickness());
-        mState.recordedMaterial[volIter->volume->geoID()].push_back(
-            std::pair(properties, rmIter->position));
-        for (int step = 1; step <= volumeStep; step++) {
+        for (int step = 0; step <= volumeStep; step++) {
```
Comment (asalzburger) [inline Core/src/Material/VolumeMaterialMapper.cpp:316]: Hi, I think the initial protection against `step==0` is that your first step might not give you a direction ... can we make a more robust way for that?

Comment (Corentin-Allaire) [reply Core/src/Material/VolumeMaterialMapper.cpp:316]: Here step==0 does not mean the first step, the step correspond to additional measurement I create between two Geantino interaction to have a more or less uniform sampling of the volume. So Step==0 correspond to the geantino interaction.
So here the direction is the direction of the geantino and there is no reason why it shouldn't have one (I think).
But maybe I should change change step to extrapolated measurement or something similar to make it clearer ?

Comment (asalzburger) [reply Core/src/Material/VolumeMaterialMapper.cpp:316]: Ok, thanks for the clarification.

Comment (Corentin-Allaire) [reply Core/src/Material/VolumeMaterialMapper.cpp:316]: Looking at it again I realise that the Direction shouldn't be computed inside the loop so I will change that quickly.

**Comment by Corentin-Allaire** [comment]
Is there a way to retrigger the checks ? I am not quite sure why the Coverage / build_debug fail...

**Pull request merged**